pyccel/ast/core.py

```diff
@@ -210,11 +210,29 @@ def broadcast(shape_1, shape_2):
             errors.report(msg,severity='fatal')
     return tuple(new_shape)
 
+def handle_precedence(args, my_precedence, commutative = False):
+    precedence = [getattr(a, 'precedence', 17) for a in args]
```

Comment:
Is it safe to use 17 as a default? Why not letting Python raise an attribute error instead?

Reply:
As per here: https://docs.python.org/3/reference/expressions.html#operator-precedence
17 is Subscription, slicing, call, attribute reference
So a Variable would have `precedence = 17`, as would an IndexedElement, and almost every other PyccelAstNode
In the syntactic step there are still a certain number of sympy objects. For example a `Variable` is not saved as such until its `dtype` is known. Thus in the syntactic stage it is still a (sympy) `Symbol`. This means that letting Python raise an attribute error will not work